src/requestHandlers/publish.ts

```diff
@@ -0,0 +1,228 @@
+        const image3Id = `${author_id}_${panel_set.id}_${panelImage3.originalname}`;
+
+        // make panels, call service as panel_set creation worked
+        const panel1 = await createPanel(sequelize, t)({
+            image:        image1Id,
+            index:        0,
+            panel_set_id: panel_set.id,
+        });
+
+        const panel2 = await createPanel(sequelize, t)({
+            image:        image2Id,
+            index:        1,
+            panel_set_id: panel_set.id,
+        });
+
+        const panel3 = await createPanel(sequelize, t)({
+            image:        image3Id,
+            index:        2,
+            panel_set_id: panel_set.id,
+        });
+
+        finalResponseObject += JSON.stringify(panel1);
+        finalResponseObject += JSON.stringify(panel2);
+        finalResponseObject += JSON.stringify(panel3);
+
+
+        // create hooks and validate
+        await Promise.all(hooks.map(async (hook) => {
+            const matchedPanel = [panel1, panel2, panel3].find(panel => panel.index === hook.panel_index);
+            if (matchedPanel === undefined) throw new Error('Hook panel_index was invalid');
+            finalResponseObject += JSON.stringify(await createHook(sequelize, t)({
+                position:          hook.position,
+                current_panel_id:  matchedPanel.id,
+                next_panel_set_id: null
+            }));
+        }));
+
+        // save to amazon 
+        const s3Image1 = await _saveImageController(image1Id, panelImage1.buffer, panelImage1.mimetype) as {id: string, } | Error;
+        if (s3Image1 instanceof Error) throw s3Image1;
+
+        const s3Image2 =  await _saveImageController(image2Id, panelImage2.buffer, panelImage2.mimetype) as {id: string, } | Error;
+        if (s3Image2 instanceof Error) throw s3Image2;
+
+        const s3Image3 = await _saveImageController(image3Id, panelImage3.buffer, panelImage3.mimetype) as {id: string, } | Error;
+        if (s3Image3 instanceof Error) throw s3Image3;
+
+
+        finalResponseObject += JSON.stringify(s3Image1);
+        finalResponseObject += JSON.stringify(s3Image2);
+        finalResponseObject += JSON.stringify(s3Image3);
+
+
+        // if gotten this far, everything worked
+        await t.commit();
+        return { success: `Panel_Set successfully published. ${finalResponseObject.toString()}` };
+    }
+    catch (err) {
+        await t.rollback();
+        return err;
+    }
+};
+
+/**
+ * Validates author_id before sending a creation request to the database
+ * @param request 
+ * @param res 
+ * @returns 
+ */
+const publish = async (request: Request, res: Response) : Promise<Response> => {
+
+    // parse the data field
+    try {
+        JSON.parse(request.body.data);
+    }
+    catch (e) {
+        return res.status(400).json({ error: 'data is not valid JSON and cannot be accepted' });
+    }
+
+    const data = JSON.parse(request.body.data);
```

Comment:
Could declaring `let data;` then defining it with `data = JSON.parse(request.body.data);` in the trycatch to avoid running `JSON.parse` twice? If having `data` as a `const` is more important then disregard this suggestion.